### PR TITLE
✨(project) import the library from ashley

### PR DESCRIPTION
**Pull request by madmatah**
The library has been extracted from the `openfun/ashley` repository in
order to use it in multiple projects. It has been renamed (the
original package name was `lti_provider`) in order to avoid confusion
with other LTI related projects.

This is basically a copy of the work done in Ashley, with our common
build process and quality check tools. The views have been simplified
and a sandbox has been added to try out the project easily.

### CODE REVIEW

File: .prettierrc [hunk 1]
```diff
@@ -0,0 +1,2 @@
+singleQuote: true
```
Comment (lunika) [inline .prettierrc]: This file can be removed

File: .circleci/config.yml [hunk 2]
```diff
@@ -0,0 +1,276 @@
...
+      - checkout
+      - run:
+          name: Check CHANGELOG max line length
+          command: |
+            # Get the longuest line width (ignoring release links)
+            test $(cat CHANGELOG.md | grep -Ev "^\[.*\]: https://github.com/openfun" | wc -L) -le 80
+
+  # ---- Backend jobs ----
+  # Build backend development environment
+  build-back:
```
Comment (lunika) [inline .circleci/config.yml]: There is no back or front application in this project, the back suffix can be removed IMO
```suggestion
  build:
```

File: .circleci/config.yml [hunk 3]
```diff
@@ -0,0 +1,276 @@
...
+            - v1-back-dependencies-{{ .Revision }}
+      - run:
+          name: Install development dependencies
+          command: pip install --user .[dev,sandbox]
+      - save_cache:
+          paths:
+            - ~/.local
+          key: v1-back-dependencies-{{ .Revision }}
+
+  lint-back:
```
Comment (lunika) [inline .circleci/config.yml]:
```suggestion
  lint:
```

File: .circleci/config.yml [hunk 4]
```diff
@@ -0,0 +1,276 @@
...
+          name: Lint code with pylint
+          command: ~/.local/bin/pylint src sandbox tests
+      - run:
+          name: Lint code with bandit
+          command: ~/.local/bin/bandit -qr src/lti_toolbox sandbox
+      - run:
+          name: Type-check code with mypy
+          command: ~/.local/bin/mypy src
+
+  test-back:
```
Comment (lunika) [inline .circleci/config.yml]:
```suggestion
  test:
```

File: .circleci/config.yml [hunk 5]
```diff
@@ -0,0 +1,276 @@
...
+      - run:
+          name: Run tests
+          command: |
+            dockerize \
+              -wait tcp://localhost:5432 \
+              -timeout 60s \
+                ~/.local/bin/pytest
+
+  # ---- Packaging jobs ----
+  package-back:
```
Comment (lunika) [inline .circleci/config.yml]:
```suggestion
  package:
```

File: README.md [hunk 6]
```diff
@@ -1,5 +1,35 @@
...
+
+- Verify LTI launch requests
+- Base views to build your own LTI launch request handlers
+- Sample Django authentication backend
+- Manage your LTI consumers from django admin
+- Demo project to quickly see it in action
+
+## Try it with our demo project !
+
+- Clone this repository (`git clone git@github.com:openfun/django-lti-toolbox.git`)
```
Comment (lunika) [inline README.md]: Should use the `https` link ?

File: docker-compose.yml [hunk 7]
```diff
@@ -0,0 +1,33 @@
+version: '3.4'
+
+services:
+  postgresql:
+    image: postgres:12
+    env_file:
+      - env.d/development/postgresql
+    ports:
```
Comment (lunika) [inline docker-compose.yml:8]: There is no need to expose this port

Comment (madmatah) [reply docker-compose.yml:8]: I leave it like this because I find it convenient to be able to access the database directly from my IDE.

**Comment by madmatah** [comment]
Thanks for your review @lunika :pray: 
I think we're OK, CI is green :wink:

**Review by lunika — APPROVED** [review]

**Pull request merged**